core/src/main/java/org/elasticsearch/common/transport/NetworkExceptionHelper.java

```diff
@@ -35,7 +36,7 @@ public static boolean isConnectException(Throwable e) {
     }
 
     public static boolean isCloseConnectionException(Throwable e) {
-        if (e instanceof ClosedChannelException) {
+        if (e instanceof ClosedChannelException || e instanceof SocketException) {
```

Comment:
I don't think that is true, SocketException is used in many different ways in the jdk we shouldn't generalize it in such a way